src/htr2hpc/api_client.py

```diff
@@ -0,0 +1,329 @@
+import datetime
+import logging
+from collections import namedtuple
+from dataclasses import dataclass
+from time import sleep
+import pathlib
+
+import humanize
+import requests
+from django.utils.text import slugify  # is django a reasonable dependency?
+
+from htr2hpc import __version__ as _version
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ResultsList:
+    """API list response."""
+
+    # all API list methods have the same structure,
+    # so use a dataclass but specify the result type
+
+    result_type: str
+    count: int
+    next: str
+    previous: str
+    results: list
+
+    def __post_init__(self):
+        # convert result entries to namedtuple class based on
+        # specified result type name
+        self.results = [to_namedtuple(self.result_type, d) for d in self.results]
+
+
+@dataclass
+class Task:
+    """API response for a task result."""
+
+    pk: int
+    document: int
+    document_part: int
+    workflow_state: int
+    label: str
+    messages: str
+    queued_at: datetime.datetime
+    started_at: datetime.datetime
+    done_at: datetime.datetime
+    method: str
+    user: int
```

Comment:
Perhaps worth some comments, especially for the cases where these are identifiers rather than contents or filenames.

Reply:
It seems like it might be worth renaming `document` to `document_id`